### PR TITLE
GH-4954 LMDB: Fix GC and record counting for deletions

### PR DESCRIPTION
**Pull request by kenwenzel**
Ensures that GC is also working if triple store needs to grow and that the correct count is returned for statement removals.

GitHub issue resolved: #4954

Briefly describe the changes proposed in this PR:

Tests removed values after the triple store's transaction is committed and correctly calls handler in removeStatement methods.
----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by hmottestad** [comment]
@kenwenzel You know you have committer rights now so you don't have to use your fork anymore.

**Comment by kenwenzel** [comment]
> @kenwenzel You know you have committer rights now so you don't have to use your fork anymore.

Thank you for the.hint. Do you favor the creation of branches in this repo rather than using a/my fork?

**Comment by hmottestad** [comment]
The failing integration tests: It's a test that is marked as disabled, but has somehow started to run again. Not sure how that happened.

**Comment by hmottestad** [comment]
> > @kenwenzel You know you have committer rights now so you don't have to use your fork anymore.
> 
> 
> 
> Thank you for the.hint. Do you prefer the creation of branches in this repo rather than using a/my fork?

Would be nice if you used the RDF4J repo. Nice thing is that it allows others to work on your branch, though that's probably not going to happen very often.

**Comment by kenwenzel** [comment]
@hmottestad Should I merge this on my own?

**Comment by hmottestad** [comment]
I prefer if you request a code review. It doesn't have to be me doing the code review, can also be someone else who is active on the project, at the moment it's mostly Jerven who is active. You can request a review from either one of us, and if you get an "approval" then feel free to merge it. 

If you don't get a response to your review request ping me and if you still don't hear anything after a few more days then feel free to do a "self review" and merge. Unless you feel that your changes are very big, a fundamental change or you have a funky solution to something that you want feedback on.

**Review by hmottestad — APPROVED** [review]
Can't say I understand much of what's going on. LGTM.

**Pull request merged**
